staging/src/k8s.io/client-go/restmapper/category_expansion.go

```diff
@@ -57,6 +57,18 @@ func NewDiscoveryCategoryExpander(client discovery.DiscoveryInterface) CategoryE
 
 // Expand fulfills CategoryExpander
 func (e discoveryCategoryExpander) Expand(category string) ([]schema.GroupResource, bool) {
+	ret, found := e.expand(category)
+	if !found {
+		if cachedDiscovery, ok := e.discoveryClient.(discovery.CachedDiscoveryInterface); ok {
```

Comment:
IMHO, these changes might be sort of hacky. `discoveryCategoryExpander` needn't aware that the discoveryClient is wrapped with cache. How about trying to reconcile the cache before the mapper retrieves the outdated result?

Reply:
🤔I am not really sure about the most elegant way but current changes doesn't look good to me. @deads2k WDYT?

Reply:
for reference, this is what the cached restmapper does:

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/restmapper/discovery.go#L222-L235

it uses cached discovery docs, and if a kind isn't found, rediscovers. that is unlikely to be appropriate here, since most things are not categories, and we don't want to force rediscovery on most calls.

Reply:
@liggitt @yue9944882 Thanks for the review.

What about current version?

When no kind is found, try expand category again.